src/GafferCortex/CompoundParameterHandler.cpp

```diff
@@ -105,33 +105,27 @@ Gaffer::Plug *CompoundParameterHandler::setupPlug( Gaffer::GraphComponent *plugP
 
 	setupPlugFlags( m_plug.get(), flags );
 
-	// remove any child plugs we don't need
-
+	// loop through the handlers and remove any that are not linked to a new parameter
+	const CompoundParameter::ParameterVector &children = m_parameter->orderedParameters();
 	std::vector<Gaffer::PlugPtr> toRemove;
-	for( Gaffer::PlugIterator pIt( m_plug.get() ); !pIt.done(); ++pIt )
+	for( HandlerMap::iterator it = m_handlers.begin(), eIt = m_handlers.end(); it!=eIt; )
 	{
-		if( (*pIt)->getName().string().compare( 0, 2, "__" ) == 0 )
-		{
-			// we leave any plugs prefixed with __ alone, on the assumption
```

Comment:
I assumed that I didn't need to consider this test anymore (about the __ plugs) because since the loop is now over the handlers, and not the plugs, we are automatically filtering out that type of plug from consideration.